### PR TITLE
Weak randomness tests for nodejs

### PR DESCRIPTION
**Pull request by CarlesDD**
## Motivation

System test for Weak randomness vulnerability detection in nodejs

## Changes

Add weak randomness endpoints in express application.
Add vulnerability location in weak randomness test for nodejs.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

[APPSEC-12060](https://datadoghq.atlassian.net/browse/APPSEC-12060)

[APPSEC-12060]: https://datadoghq.atlassian.net/browse/APPSEC-12060?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by simon-id** [comment]
1. do we not need to enable something in the manifest ?
2. why not implement in the other weblog variants ?

**Comment by CarlesDD** [comment]
I left `missing_feature` in manifest because the feature has not been released yet. My plan is to merge this PR without manifest modifications and once the feature has been released open a new PR pinning the tracer version in the manifest.

Regarding weblog variants, I've added the endpoints for this test in the `express4-typescript` one.

**Review by uurien — APPROVED** [review]

File: utils/build/docker/nodejs/express4-typescript/iast.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    app.get('/iast/weak_randomness/test_insecure', (req: Request, res: Response) => {
+        const randomNumber = Math.random()
+        res.send(`OK:${randomNumber}`)
+    })
+
+    app.get('/iast/weak_randomness/test_secure', (req: Request, res: Response) => {
+        const randomBytes = crypto.randomBytes(256).toString('hex')
+        res.send(`OK:${randomBytes}`)
+    })
+}
```
Comment (simon-id) [inline utils/build/docker/nodejs/express4-typescript/iast.ts:17]: whoops looks like we're lacking some linting here
```suggestion
module.exports = function (app: Express) {
  app.get('/iast/weak_randomness/test_insecure', (req: Request, res: Response) => {
    const randomNumber = Math.random()
    res.send(`OK:${randomNumber}`)
  })

  app.get('/iast/weak_randomness/test_secure', (req: Request, res: Response) => {
    const randomBytes = crypto.randomBytes(256).toString('hex')
    res.send(`OK:${randomBytes}`)
  })
}
```

Comment (uurien) [reply utils/build/docker/nodejs/express4-typescript/iast.ts]: Future work: add lint task in typescript project

**Review by simon-id — APPROVED** [review]

**Pull request merged**